docs/src/_contribute/cli.md

```diff
@@ -0,0 +1,247 @@
+that a users experiences when browsing the docs or experimenting with the CLI. As such, we aim to provide a clear and
+consistent interface that is easy to use. This guide aims to codify the design guidelines for the CLI so that
+contributors can easily update or extend the CLI with new features, commands, options, and enhancements.
+
+This document is somewhat aspirational and a work in progress. Portions of the CLI may violate the design and style
+guidelines, but we aim to update the CLI overtime to ensure that it is as consistent as possible.
+
+### Consistent definition of Groups, Commands, Sub-commands, Arguments, and Options
+
+Groups, Commands, Sub-commands, Arguments, Options, and Flags often have slightly different meanings for users. So are
+defined as follows for use in `meltano`.
+
+- Group is a group of commands that are related to a specific area of the CLI, typically a specific meltano feature.
+  For example, the `meltano schedule` feature-group contains commands related to managing Meltano schedules.
+- command/sub_command are used to perform specific tasks or used to group a set of command around a specific feature
+  or task. Sub-commands take the form of parameters, but require additional arguments or options themselves.
+- Arguments are positional parameters that are passed to a command. Arguments do not require additional options or
+  arguments - otherwise they would be considered a sub-command.
+- Options (switch, option flags, or flags) are options that alter the behavior (e.g. `--dry-run/--verbose`) or named
+  input options (e.g. `--tasks=`). They come in two forms. A long form: `--option-name`, and a short form: `-o`.
+
+Given a command like `meltano job set JOB_NAME --tasks=[<task>...]`:
+
+- `meltano` is the global click command group for Meltano. This is the main entry point to the CLI. When we refer to global options this is the level we refer to.
+- `job` is the click command group for the `job` command. This is the main command group for the `job` command. This is typically the top level for a Meltano feature.
+- `set` is a `sub-command` of the `job` command.
+- `JOB_NAME` is an `argument` of the `set` sub-command.
+- `--tasks` is named `option` of the `set` sub-command.
+
+For a technical explanation of how commands and groups work see https://click.palletsprojects.com/en/8.0.x/commands/
+
+### Use of short and long options
+
+The options you expect to be used most frequently should always include both the long AND short variation of the option.
+For example `--help` and `-h`, `--tasks` and `-t`, `--force` and `-f`.
+
+
+### Desired verb/command linkage and structure
+
+1. Organize in a command group around a feature below the top level where possible.
+2. Coalesce commands that operate on the same resource of a feature as a group of sub-commands.
+3. Use transitive verbs like `delete`, `set`, `list`, `get` as sub-commands to perform work within a group where appropriate.
```

Comment:
I think we are preferring `remove` over `delete` so it might be good to adjust the samples here slightly.

Related: I wonder if it would be helpful to document the common/preferred CRUD-esque verbs: 

- `add` and `remove` to create or remove a construct like "schedules" or "jobs". (Maybe also `set` to override/update after creation, as in `meltano job JOB_NAME set ...`.)
- `get`, `set`, `list`, `clear` for fetching and setting artifacts like "state" and "secrets".
- `list` to enumerate items.
- `describe` to show details about an item.

Reply:
👍 Swapped in `remove` and add a standardized verbs section.

Reply:
Looks great. Thanks!